mod/groups/actions/groups/edit.php

```diff
@@ -19,6 +19,14 @@ function profile_array_decoder(&$v) {
 foreach (elgg_get_config('group') as $shortname => $valuetype) {
 	$input[$shortname] = get_input($shortname);
 
+	if ($input[$shortname] === '') {
+		// The group profile displays all profile fields that have a value.
+		// We don't want to display fields with empty string value, so we
+		// prevent it getting saved by setting the value to null.
+		$input[$shortname] = null;
```

Comment:
ElggEntity in fact first removes all metadata: https://github.com/Elgg/Elgg/blob/1.10/engine/classes/ElggEntity.php#L412

Metadata service later ignores the null value: https://github.com/Elgg/Elgg/blob/1.10/engine/classes/Elgg/Database/MetadataTable.php#L123